oci/pkg/oci/image.go

```diff
@@ -0,0 +1,165 @@
+package oci
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/containerd/containerd/remotes/docker"
+	orascontent "github.com/deislabs/oras/pkg/content"
+	orascontext "github.com/deislabs/oras/pkg/context"
+	"github.com/deislabs/oras/pkg/oras"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func init() {
+	orascontext.GetLogger(context.Background()).Logger.SetLevel(logrus.ErrorLevel)
+}
+
+// ImageReference is a generic wrapper around the parts of an OCI image name. Not all of the fields exist for every
+// image.
+type ImageReference struct {
+	FullName string
+	Name     string
+	Registry string
+	Hash     string
+	Tag      string
+}
+
+// ValidateImageName will return nil if name is a valid, fully qualified image reference. This is based on
+// https://github.com/helm/helm/blob/7ffc879f137bd3a69eea53349b01f05e3d1d2385/internal/experimental/registry/reference.go#L48
+func ValidateImageName(name string) (*ImageReference, error) {
+	if name == "" {
+		return nil, fmt.Errorf("%s is not a valid image name", name)
+	}
+
+	return getNameParts(name)
+}
+
+// getNameParts will attempt to return a { "hostname/image-name", "tag or sha" } list of the provided image name. There maybe less than the full 2 elements if one of the parts isn't included in the original name.
+func getNameParts(name string) (*ImageReference, error) {
```

Comment:
The go-containerregistry library has a pretty solid set of methods for this kind of name parsing, I'd love to avoid having to maintain it in two places.

https://godoc.org/github.com/google/go-containerregistry/pkg/name#ParseReference